dipy/core/gradients.py

```diff
@@ -67,6 +69,71 @@ def __init__(self, gradients, big_delta=None, small_delta=None,
         self.big_delta = big_delta
         self.small_delta = small_delta
         self.b0_threshold = b0_threshold
+        if isinstance(btens, str):
+            b_tensors = np.zeros((len(self.bvals), 3, 3))
+            if btens == 'LTE':
+                b_tensor = np.array([[1, 0, 0],
+                                     [0, 0, 0],
+                                     [0, 0, 0]])
```

Comment:
These three matrice are used at least twice here. Can we think of defining them only once and re-using the object?

Reply:
Done.

Reply:
May be a separate function to generate them depending on the encoding type would be more appropriate/easier to re-use, maintain and expand, but I am fine with the current solution for now. Thanks @kerkelae.